jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2818,11 +2818,62 @@ public static MethodHandle dropArguments(MethodHandle originalHandle, int locati
 	 * @param originalHandle the original method handle to be transformed
 	 * @param skippedArgumentCount the number of argument to be skipped from the original method handle
 	 * @param valueTypes a List of the argument types to be inserted
-	 * @param location the location of the first argument to be removed
+	 * @param location the (zero-indexed) location of the first argument to be removed
 	 * @return a MethodHandle representing a transformed handle as described above
 	 */
-	public static MethodHandle dropArgumentsToMatch(MethodHandle originalHandle, int skippedArgumentCount, List<Class<?>> valueTypes, int location) {
-		throw new UnsupportedOperationException("The method has not yet been implemented for now"); //$NON-NLS-1$
+	public static final  MethodHandle dropArgumentsToMatch(MethodHandle originalHandle, int skippedArgumentCount, List<Class<?>> valueTypes, int location) {
+		/* implicit null checks */
+		MethodType originalType = originalHandle.type;
+		Class<?>[] valueTypesCopy = new Class<?>[valueTypes.size()];
+
+		/* check if indexing is in range */
+		if ((0 > skippedArgumentCount) ||
+			(skippedArgumentCount > originalType.parameterCount())) {
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0670")); //$NON-NLS-1$
+		}
+		if ((0 > location) ||
+			(valueTypes.size() < location) ||
+			(valueTypes.size() < location + originalType.parameterCount() - skippedArgumentCount)) {
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0671")); //$NON-NLS-1$
+		}
+
+		/* check for void.class in list during clone process */
+		for (int i = 0; i < valueTypes.size(); i++) {
+			if (valueTypes.get(i) == void.class) {
+				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0672")); //$NON-NLS-1$
```

Comment:
This should be proceeded by a comment that includes the K0670 and the message using preprocessor tags.  Search for other uses of the com.ibm.oti.util.Msg class to see how the comment should look